### PR TITLE
chore: add rollup bundling for circuit-lib.js package.

### PR DESCRIPTION
**Pull request by sergeytimoshin**

### CODE REVIEW

File: circuit-lib/circuit-lib.js/src/elgamal-babyjubjub/elgamal.ts [hunk 1]
```diff
@@ -168,6 +167,16 @@ namespace ElGamalUtils {
   }
 }
 
+/**
+ * @param plaintext A 32-bit bigint
+ * @returns A point on the Baby Jubjub curve
+ */
+export function encode(plaintext: bigint): ExtPointType {
+  if (plaintext >= BigInt(2 ** 32)) {
+    throw new Error("The plaintext should nit be bigger than a 32-bit bigint");
```
Comment (SwenSchaeferjohann) [inline circuit-lib/circuit-lib.js/src/elgamal-babyjubjub/elgamal.ts:176]: small typo 
```suggestion
    throw new Error("The plaintext should not be bigger than a 32-bit bigint");
```

File: circuit-lib/circuit-lib.js/src/elgamal-babyjubjub/pointEncoding.ts [hunk 2]
```diff
@@ -3,16 +3,6 @@ import { ExtPointType } from "@noble/curves/abstract/edwards";
...
-/**
- * @param plaintext A 32-bit bigint
- * @returns A point on the Baby Jubjub curve
- */
-export function encode(plaintext: bigint): ExtPointType {
-  if (plaintext >= BigInt(2 ** 32)) {
-    throw new Error("The plaintext should nit be bigger than a 32-bit bigint");
-  } else return babyjubjubExt.BASE.multiplyUnsafe(plaintext);
-}
-
```
Comment (SwenSchaeferjohann) [inline circuit-lib/circuit-lib.js/src/elgamal-babyjubjub/pointEncoding.ts:15]: what's the reasoning for moving encode to elgamal.ts?

File: pnpm-lock.yaml [hunk 3]
```diff
@@ -85,7 +85,7 @@ importers:
         version: 10.0.0(mocha@10.2.0)
       ts-node:
         specifier: ^10.9.1
-        version: 10.9.1(@types/node@20.10.4)(typescript@5.3.2)
+        version: 10.9.1(@types/node@20.10.4)(typescript@4.9.5)
```
Comment (SwenSchaeferjohann) [inline pnpm-lock.yaml:88]: is there a particular reason why it's downgrading to ts4.9?

**Review by SwenSchaeferjohann — APPROVED** [review]
awesome, just one nit and a few questions; besides that lgtm - already approving @sergeytimoshin

File: circuit-lib/circuit-lib.js/src/elgamal-babyjubjub/pointEncoding.ts (hunk 2 above)
Comment (sergeytimoshin) [reply circuit-lib/circuit-lib.js/src/elgamal-babyjubjub/pointEncoding.ts:15]: Circular dependencies. 

I'm looking at this with fresh eyes and thinking that this is probably not the best (semantically) point to move this thing, I'll do one more tiny refactoring here.

**Review by ananas-block — COMMENTED** [review]
Haven't looked much into bundling thus leaving you guys to reason about this @sergeytimoshin @SwenSchaeferjohann .

**Pull request closed without merging**